Assets/Scripts/GameManagement/DataPersistence/DataPersistenceManager.cs

```diff
@@ -0,0 +1,96 @@
+using Inventory;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.InputSystem;
+using static UnityEngine.InputSystem.InputAction;
+
+public class DataPersistenceManager
+{
+    private static DataPersistenceManager instance = null;
+    private static readonly object instanceLock = new object(); //thread-safe for co-routines
+
+    private InputAction saveAction;
+    private InputAction loadAction; //only needed for testing
+
+    public event Action onSaveTriggered;
+    public event Action onLoadTriggered;
+
+    private FileDataHandler fileDataHandler;
+    private string fileName = "AngelInsolventSaveFile.txt";
+
+    DataPersistenceManager()
+    {
+        this.fileDataHandler = new FileDataHandler(Application.persistentDataPath, fileName);
+        saveAction = new InputAction(binding: "<Keyboard>/f5"); //should make this serializable somewhere for multiple platforms' inputs
+        saveAction.performed += SaveGame;
+        saveAction.Enable();
+
+        loadAction = new InputAction(binding: "<Keyboard>/f6");
+        loadAction.performed += LoadGame;
+        loadAction.Enable();
```

Comment:
Code does not use InputAction asset from new input system or unbind actions with destructor, might be ok if these are only for testing.

Reply:
Will update these to use new input system; the load one is only for testing but I planned to have the save one in the actual game

Reply:
Make sure to unbind anything you plan on keeping for the final game as well just to be safe, this syntax could be helpful: https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/finalizers

Reply:
fo sure thank u